keps/sig-cluster-lifecycle/0015-kubeadm-join-master.md

```diff
@@ -0,0 +1,431 @@
+      in `kube-system` namespace.
+
+      > This requires to grant access to the above configMap for
+      > `system:bootstrappers` group.
+
+   2. Check if the cluster is ready for joining a new master node:
+
+      a. Check if the cluster has a stable `controlplaneAddress`
+      a. Check if the cluster uses an external etcd
+      a. Checks if the mandatory certificates exists on the file system
+
+   3. Prepare the node for joining as a master node:
+
+      a. Create missing certificates (in any).
+         > please note that by creating missing certificates kubeadm can adapt seamlessly
+         > to a dynamic workflow or to a static workflow (and to apiserver advertise address
+         > of the joining node). see following paragraphs for more details for additional info.
+
+      a. In case of control plane deployed as static pods, create related kubeconfig files
+      and static pod manifests.
+
+      > see "Strategies for deploying control plane components" paragraph
+      > for additional info about this step.
+
+   4. Create the admin.conf kubeconfig file
+
+3. Executes the kubelet TLS bootstrap process [No changes to this step]:
+
+4. In case of `join --master` [New step]
+
+   1. Apply master taint and label to the node.
+
+      > This action is executed using the admin.conf identity created above;
+
+   2 Update the `kubeadm-config` configMap with the information about the new master node
+
+#### dynamic workflow (advertise-address == `controlplaneAddress`)
+
+There are many ways to configure an highly available cluster.
+
+Among them, the approach best suited for a dynamic bootstrap workflow requires the
+user to set the `--apiserver-advertise-address` of each master, including the bootstrap master
+itself, equal to the `controlplaneAddress` endpoint provided during kubeadm init
+(the IP/DNS of the external load balancer).
+
+By using the same advertise address for all the IP masters, `kubeadm init` can create
+a unique API server serving certificate that could be shared across many masters nodes;
+no changes will be required to this certificate when adding/removing master nodes.
+
+Please note that:
+
+- if the user is not planning to distribute the apiserver serving certificate among masters,
+  kubeadm will generate a new apiserver serving certificate “almost equal” to the certificate
+  created on the bootstrap master (it differs only for the domain name of the joining master)
+
+#### Static workflow (advertise-address != `controlplaneAddress`)
+
+In case of a static bootstrap workflow the final layout of the controlplane - the number, the
+name and the IP of master nodes - is know in advance.
+
+Given such information, the user can choose a different approach where each master has a
+specific apiserver advertise address different from the `controlplaneAddress`.
+
+Please note that:
+
+- if the user is not planning to distribute the apiserver certificate among masters, kubeadm
```

Comment:
I don't think the user ever should distribute/share the API server serving cert

Reply:
see to following comment